### PR TITLE
Add basic info about handling transaction-action-request webhook

### PR DESCRIPTION
**Pull request by korycins**
Add some basic info about a new webhook that is triggered when a staff user or app is requesting some action on a transaction.

### CODE REVIEW

**Comment by patrys** [comment]
I think it's time we split https://saleor-docs-6i7u5r4oj-saleorcommerce.vercel.app/docs/3.x/developer/checkout, turn it into a category in the sidebar, leave only the intro in this document, and move every section into its own document under `docs/developer/checkout/*.mdx`.

**Review by timuric — APPROVED** [review]

File: docs/developer/checkout.mdx [hunk 1]
```diff
@@ -1147,9 +1200,204 @@ As a result, we get an order object:
...
+    transaction {
+      id
+    }
+  }
+}
+```
+
+:::
+
+### Handling an action request for transaction
```
Comment (karolkielecki) [inline docs/developer/checkout.mdx:1373]:
```suggestion
### Handling an action request for a transaction
```

File: docs/developer/checkout.mdx [hunk 2]
```diff
@@ -1147,9 +1200,204 @@ As a result, we get an order object:
...
+
+Action request is called when a staff user or an app triggers a request of action that should be called for a given transaction.
```
Comment (karolkielecki) [inline docs/developer/checkout.mdx:1375]:
```suggestion
An action request is called when a staff user or an app triggers a request of an action that should be called for a given transaction.
```

File: docs/developer/checkout.mdx [hunk 3]
```diff
@@ -1147,9 +1200,204 @@ As a result, we get an order object:
...
+If your app should also process payment request actions (like charge, refund, or void) triggered by staff users on
```
Comment (karolkielecki) [inline docs/developer/checkout.mdx:1376]:
```suggestion
If your app should also process payment request actions (like charge, refund, or void) triggered by staff users on the 
```

File: docs/developer/checkout.mdx [hunk 4]
```diff
@@ -1147,9 +1200,204 @@ As a result, we get an order object:
...
+Saleor dashboard side or by app, make sure that your app is subscribed to
```
Comment (karolkielecki) [inline docs/developer/checkout.mdx:1377]:
```suggestion
Saleor dashboard side or by the app, make sure that your app is subscribed to the
```

File: docs/developer/checkout.mdx [hunk 5]
```diff
@@ -1147,9 +1200,204 @@ As a result, we get an order object:
...
+[transaction-action-request](developer/extending/apps/sample-webhook-payloads.mdx#transaction-action-request) webhook.
+
+The below chart shows a workflow for handling `refund` request.
```
Comment (karolkielecki) [inline docs/developer/checkout.mdx:1380]:
```suggestion
The chart below shows a workflow for handling a `refund` request.
```

File: docs/developer/checkout.mdx [hunk 6]
```diff
@@ -1147,9 +1200,204 @@ As a result, we get an order object:
...
+    App -->>- Saleor: OK
+    App ->>+ PaymentProvider: Execute refund <br>action
+    PaymentProvider -->>- App: Refunded
+    App ->>+ Saleor: Call transactionUpdate <br>to update transaction with<br> a result recieved after <br>requested action
+    Saleor -->>- App: Updated transaction
+```
+
 ## Removing old checkouts
+
 In order not to overflow the database, unfinished checkouts are deleted.
```
Comment (karolkielecki) [inline docs/developer/checkout.mdx:1398]:
```suggestion
In order not to overload the database, unfinished checkouts are deleted.
```

File: docs/developer/extending/apps/sample-webhook-payloads.mdx [hunk 7]
```diff
@@ -2160,5 +2160,42 @@ Each webhook payload has `notify_event` field which defines the type of the noti
    "product_slug": "orange-juice-500-ml"
 }
 
+```
+</details>
+
+## TRANSACTION ACTION request
+
+`TRANSACTION_ACTION_REQUEST` is called after requesting a change in transaction state (called by staff user or by an app).
```
Comment (karolkielecki) [inline docs/developer/extending/apps/sample-webhook-payloads.mdx:2168]:
```suggestion
`TRANSACTION_ACTION_REQUEST` is called after requesting a change in transaction state (called by a staff user or by an app). 
```

File: docs/developer/extending/apps/sample-webhook-payloads.mdx [hunk 8]
```diff
@@ -2160,5 +2160,42 @@ Each webhook payload has `notify_event` field which defines the type of the noti
...
 }
 
+```
+</details>
+
+## TRANSACTION ACTION request
+
+`TRANSACTION_ACTION_REQUEST` is called after requesting a change in transaction state (called by staff user or by an app). 
+The `action` object contains all details related to the requested action. 
+The `transaction` contains all details related to `TransactionItem` object for which the requested action should be called.
```
Comment (karolkielecki) [inline docs/developer/extending/apps/sample-webhook-payloads.mdx:2170]:
```suggestion
The `transaction` contains all details related to the `TransactionItem` object for which the requested action should be called.
```

File: docs/developer/checkout.mdx [hunk 9]
```diff
@@ -1071,8 +1078,54 @@ A response would look like this:
...
-## Creating order from checkout
+This feature is currently in **Feature Preview**. That means this part of Saleor is not complete
+but is available to experiment with and provide feedback, and is subject to change.
+
+:::
+
+The flow where the payment process is handled by an external Saleor App.
+Saleor expects to receive a result of the payment transaction processed by Saleor App.
+
+The below charts show the workflow of a checkout process.
```
Comment (karolkielecki) [inline docs/developer/checkout.mdx:1093]:
```suggestion
The charts below show the workflow of a checkout process.
```

**Review by karolkielecki — CHANGES_REQUESTED** [review]

**Comment by timuric** [comment]
@korycins grammar improvements were added, please resolve conflict so we can merge

**Comment by korycins** [comment]
@timuric Done

**Comment by peelar** [comment]
@korycins What's the situation here? Do we still need it? If so, please resolve the conflicts and I will handle the rest.

**Comment by korycins** [comment]
@peelar no more conflicts 🤗

**Review by peelar — APPROVED** [review]

**Pull request merged**